src/__internal__/input/input.test.tsx

```diff
@@ -0,0 +1,258 @@
+import React from "react";
+import { render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import Input, { EnterKeyHintTypes } from "./input.component";
+import { InputContext } from "../input-behaviour";
+
+test("renders an input element with type 'text'", () => {
+  render(<Input />);
+
+  const input = screen.getByRole("textbox");
+  expect(input).toBeVisible();
+});
+
+test("focuses the input element when rendered if the `autoFocus` prop is true", () => {
+  render(<Input autoFocus />);
+
+  const input = screen.getByRole("textbox");
+  expect(input).toHaveFocus();
+});
+
+test.each([
+  "enter",
+  "done",
+  "go",
+  "next",
+  "previous",
+  "search",
+  "send",
+] as EnterKeyHintTypes[])(
+  "'enterKeyHint' is correctly passed to the input when prop value is %s",
+  (keyHint) => {
+    render(<Input enterKeyHint={keyHint} />);
+
+    const input = screen.getByRole("textbox");
+    expect(input).toHaveAttribute("enterKeyHint", keyHint);
+  }
+);
+
+test("should invoke the `inputRef` callback from the input context provider, when the input is rendered", () => {
+  const inputRef = jest.fn();
+  render(
+    <InputContext.Provider value={{ inputRef }}>
+      <Input />
+    </InputContext.Provider>
+  );
+
+  expect(inputRef).toHaveBeenCalled();
+});
+
+test("triggers a passed function via the `onFocus` prop when the input is focused", () => {
+  const onFocusMock = jest.fn();
+  render(<Input onFocus={onFocusMock} />);
+
+  const input = screen.getByRole("textbox");
+  input.focus();
+
+  expect(onFocusMock).toHaveBeenCalled();
+});
+
+test("triggers a passed function via the `onFocus` prop from the input context provider, when the input is focused", () => {
+  const onFocusMock = jest.fn();
+  render(
+    <InputContext.Provider value={{ onFocus: onFocusMock }}>
+      <Input />
+    </InputContext.Provider>
+  );
+
+  const input = screen.getByRole("textbox");
+  input.focus();
+
+  expect(onFocusMock).toHaveBeenCalled();
+});
+
+test("does not throw an error when the input is focused if `no onFocus` is defined", async () => {
+  render(<Input />);
+
+  const user = userEvent.setup();
+  const input = screen.getByRole("textbox");
+
+  expect(async () => user.click(input)).not.toThrow();
+});
```

Comment:
**issue**: it seems this test, along with the `onBlur` equivalent on line 113, is causing [an overlapping `act` errors in the console](https://github.com/Sage/carbon/actions/runs/10992283725/job/30516394996#step:5:956).

Additionally, these two tests might fail correctly if an error is thrown. In Jest, the [function that raises the error needs to be passed directly to `expect` within a wrapper function](https://jestjs.io/docs/expect#tothrowerror), like this:

```tsx
const handleFocus = () => {
   throw new Error("Something bad happened in handleFocus");
}

expect(() => handleFocus()).toThrow();
```

Since there doesn't seem to be any error-prone code within `Input`'s `handleFocus` and `handleBlur` functions, it might be worth reconsidering whether the error tests are necessary. What are your thoughts on removing these?